app/flows/initial_request_flow.rb

```diff
@@ -189,6 +194,18 @@ def training_provider
                            end
   end
 
+  def invalid_search_query?
+    return @invalid_search_query if @invalid_search_query
```

Comment:
I'm trying to move the validation logic into the form object where i think it better belongs. see https://github.com/DFE-Digital/publish-teacher-training/pull/1123/files

Reply:
Ah brilliant - form object sounds like a good idea. Was going to ask about his possibly refactoring. 
I'll rebase onto your changes 👍